packages/@aws-cdk/aws-certificatemanager/lib/dns-validated-certificate.ts

```diff
@@ -0,0 +1,94 @@
+import { CustomResource } from '@aws-cdk/aws-cloudformation';
+import { PolicyStatement } from '@aws-cdk/aws-iam';
+import { AssetCode, Function, Runtime } from '@aws-cdk/aws-lambda';
+import { IHostedZone } from '@aws-cdk/aws-route53';
+import { Construct, Output } from '@aws-cdk/cdk';
+import path = require('path');
+import { CertificateImportProps, CertificateProps, ICertificate } from './certificate';
+
+export interface DNSValidatedCertificateProps extends CertificateProps {
+    /**
+     * Route 53 Hosted Zone used to perform DNS validation of the request.  The zone
+     * must be authoritative for the domain name specified in the Certificate Request.
+     */
+    hostedZone: IHostedZone;
+}
+
+/**
+ * A certificate managed by AWS Certificate Manager.  Will be automatically
+ * validated using DNS validation against the specified Route 53 hosted zone.
+ */
+export class DNSValidatedCertificate extends Construct implements ICertificate {
+    public readonly certificateArn: string;
+    private normalizedZoneName: string;
+    private hostedZoneId: string;
+    private domainName: string;
+
+    constructor(scope: Construct, id: string, props: DNSValidatedCertificateProps) {
+        super(scope, id);
+
+        this.domainName = props.domainName;
+        this.normalizedZoneName = props.hostedZone.zoneName;
+        // Remove trailing `.` from zone name
+        if (this.normalizedZoneName.endsWith('.')) {
+            this.normalizedZoneName = this.normalizedZoneName.substring(0, this.normalizedZoneName.length - 1);
+        }
+
+        // Remove any `/hostedzone/` prefix from the Hosted Zone ID
+        this.hostedZoneId = props.hostedZone.hostedZoneId;
+        if (this.hostedZoneId.startsWith('/hostedzone/')) {
+            this.hostedZoneId = this.hostedZoneId.replace(/^\/hostedzone\//, '');
```

Comment:
In fact you can always do this, don't need to wrap it in an `if`.

Reply:
Fixed

Reply:
Fixed